service-search-nls/src/main/java/fi/nls/oskari/search/util/ScoredSearchHit.java

```diff
@@ -3,14 +3,20 @@
 class ScoredSearchHit implements Comparable<ScoredSearchHit> {
     public final String text;
     public final int score;
+    public final Boolean isExact;
 
-    public ScoredSearchHit(String t, int s) {
+    public ScoredSearchHit(String t, int s, boolean e) {
         text = t;
         score = s;
+        isExact = e;
     }
 
     @Override
     public int compareTo(ScoredSearchHit other) {
-        return score - other.score;
+        int comparison = this.isExact.compareTo(other.isExact);
+        if (comparison == 0) {
+            return score - other.score;
```

Comment:
Integer.compare(int a, int b); would handle edge cases better, most probably won't be a problem here, but for future reference.